mentat/code_context.py

```diff
@@ -271,37 +273,133 @@ def _get_all_features(
                     all_features.append(full_feature)
                 else:
                     _split_features = split_file_into_intervals(
-                        full_feature, user_features=self.include_files.get(abs_path, [])
+                        full_feature, user_features=self.include_files.get(path, [])
                     )
                     all_features += _split_features
             else:
                 _feature = CodeFeature(
-                    abs_path, level=level, diff=diff_target, user_included=user_included
+                    path, level=level, diff=diff_target, user_included=user_included
                 )
                 all_features.append(_feature)
 
         return sorted(all_features, key=lambda f: f.path)
 
-    def include_file(self, path: Path):
-        paths, invalid_paths = get_include_files([path], [])
-        for new_path, new_features in paths.items():
-            if new_path not in self.include_files:
-                self.include_files[new_path] = []
-            for feature in new_features:
-                self.include_files[new_path].append(feature)
-        return list(paths.keys()), invalid_paths
-
-    def exclude_file(self, path: Path):
-        # TODO: Using get_include_files here isn't ideal; if the user puts in a glob that
-        # matches files but doesn't match any files in context, we won't know what that glob is
-        # and can't return it as an invalid path
-        paths, invalid_paths = get_include_files([path], [])
-        removed_paths = list[Path]()
-        for new_path in paths.keys():
-            if new_path in self.include_files:
-                removed_paths.append(new_path)
-                del self.include_files[new_path]
-        return removed_paths, invalid_paths
+    def include(
+        self, path: Path | str, ignore_patterns: Iterable[Path | str] = []
+    ) -> Set[Path]:
+        """Add code to the context
+
+        '.' is replaced with '*' (recusively search the cwd)
```

Comment:
I have this logic here since I believe this is what people are meaning when they type "."

Reply:
Don't we already recursively search directories, so shouldn't "." already work like that? Let's not replace it with "*".

Reply:
"." for git recursively searches, but it does not for glob patterns.

Reply:
But "." should point to a directory (the cwd) which assuming we didn't change how including directories works should auto include all sub files and subdirectories

Reply:
Ahh I see what you mean now. Yeah I'll remove this